lib/bump/update_checkers/python.rb

```diff
@@ -1,15 +1,20 @@
 # frozen_string_literal: true
 require "excon"
 require "bump/update_checkers/base"
+require "bump/shared_helpers"
 
 module Bump
   module UpdateCheckers
     class Python < Base
       def latest_version
         @latest_version ||=
           begin
-            url = "https://pypi.python.org/pypi/#{dependency.name}/json"
-            JSON.parse(Excon.get(url).body)["info"]["version"]
+            pypi_response = Excon.get(
+              dependency_url,
+              middlewares: SharedHelpers.excon_middleware
+            )
+
+            JSON.parse(pypi_response.body)["info"]["version"]
```

Comment:
Hmmm, just had a look through the code and Python and Node are both lacking any kind of "this dependency actually exists" check.

I could use `dig` above, but I think it would be better to put a separate PR in to check existence explicitly. Then, if the dependency exists but we get some malformed stuff back from PyPi / NPM, we probably *do* want to explode.